### PR TITLE
feat: add venice.ai api model provider

### PR DESCRIPTION
**Pull request by proteanx**
integrates venice api

# Risks

Low, just an additional API option (that's really fast) 

# Background

Venice.ai is an artificial intelligence platform that provides access to advanced AI models, prioritizing user privacy by storing data only in the user's browser. It offers a paid version, Venice Pro, with higher limits and customization options, which is also available for free to users holding 1 Morpheus (MOR) token. The platform supports document uploads and image generation, and has a presence on Twitter, Farcaster, and Instagram.

## What does this PR do?

Adds the Venice AI API for accessing its supported models (really fast)

## What kind of change is this?

Feature (non-breaking change which adds functionality)

# Documentation changes needed?

My changes do not require a change to the project documentation. Usage is pretty straightforward.

# Testing

## Where should a reviewer start?

I tested with the default characters (by changing API option to venice)

## Detailed testing steps

Configure VENICE_API_KEY in .env
Change a character json to use modelProvider "venice"
Test text generation using configured model ```pnpm start --characters=/path/to/chararacter```

## Discord username

same as github: @proteanx

### CODE REVIEW

File: packages/core/src/models.ts [hunk 1]
```diff
@@ -451,6 +451,21 @@ export const models: Models = {
...
+            stop: [],
+            maxInputTokens: 128000,
+            maxOutputTokens: 8192,
+            temperature: 0.6,
+        },
+        model: {
+            [ModelClass.SMALL]: "llama-3.3-70b",
+            [ModelClass.MEDIUM]: "llama-3.3-70b",
+            [ModelClass.LARGE]: "llama-3.1-405b",
+            [ModelClass.IMAGE]: "fluently-xl",
```
Comment (oxSaturn) [inline packages/core/src/models.ts:466]: Maybe allow users to configure them through .env? E.g., https://github.com/ai16z/eliza/pull/999/files

Comment (proteanx) [reply packages/core/src/models.ts]: good call and done!

**Comment by proteanx** [comment]
I didn't test the image generation and it looks like it will require more work so I removed it for now and will make a new PR to add it later.

It also appears venice doesn't use frequency or presence penalty parameters (yet at least) so I removed that (for now).

This PR should be good now

**Review by odilitime — APPROVED** [review]

**Comment by proteanx** [comment]
@odilitime should I push the pnpm-lock.yaml back in to fix merge conflict?

File: packages/core/src/generation.ts [hunk 2]
```diff
@@ -479,6 +479,29 @@ export async function generateText({
                 break;
             }
 
+            case ModelProviderName.VENICE: {
```
Comment (ebrakke) [inline packages/core/src/generation.ts:482]: You can just add this as a case around line 149, as venice follow the openAI api spec.

File: packages/core/src/models.ts [hunk 3]
```diff
@@ -451,6 +451,20 @@ export const models: Models = {
...
+        settings: {
+            stop: [],
+            maxInputTokens: 128000,
+            maxOutputTokens: 8192,
+            temperature: 0.6,
+        },
+        model: {
+            [ModelClass.SMALL]: settings.SMALL_VENICE_MODEL || "llama-3.3-70b",
+            [ModelClass.MEDIUM]: settings.MEDIUM_VENICE_MODEL || "llama-3.3-70b",
+            [ModelClass.LARGE]: settings.LARGE_VENICE_MODEL || "llama-3.1-405b",
```
Comment (ebrakke) [inline packages/core/src/models.ts:465]: Could probably add an image model here like `flux-dev-unsencored` as the default with the option to control in the .env file.

Comment (proteanx) [reply packages/core/src/models.ts:465]: I originally had image generation in here but it needs more work so I figured I'd add it in a later PR. In the first commit here it's included though.

File: packages/core/src/generation.ts (hunk 2 above)
Comment (proteanx) [reply packages/core/src/generation.ts:482]: Yea I tested this as well (definitely works) but I plan to add the venice_parameters that are exclusive to venice as they come out.

**Comment by lalalune** [comment]
Merging this so we can stay up to date, expecting there's another PR coming for more Venice stuff. Great work on this.

**Pull request merged**